docs/user-guide/horizontal-pod-autoscaling/index.md

```diff
@@ -79,55 +81,35 @@ i.e. you cannot bind a Horizontal Pod Autoscaler to a replication controller and
-
-```yaml
-    ports:
-    - hostPort: 8080
-      containerPort: 8080
-```
-
-### Specifying target
-
-HPA for custom metrics is configured via an annotation. The value in the annotation is interpreted as a target metric value averaged over
-all running pods. Example: 
-
-```yaml
-    annotations:
-      alpha/target.custom-metrics.podautoscaler.kubernetes.io: '{"items":[{"name":"qps", "value": "10"}]}'
-```
+## Support for custom metrics
 
-In this case, if there are four pods running and each pod reports a QPS metric of 15 or higher, horizontal pod autoscaling will start two additional pods (for a total of six pods running).
+**Note**: Kubernetes 1.2 added alpha support for scaling based on application-specific metrics using special annotations.
+Support for this was removed in Kubernetes 1.6 in favor of the `autoscaling/v2alpha1` API.  While the old method for collecting
+custom metrics is still available, these metrics will not be available for use by the Horizontal Pod Autoscaler, and the former
+annotations for specifying which custom metrics to scale on are no longer honored by the Horizontal Pod Autoscaler controller.
 
-If you specify multiple metrics in your annotation or if you set a target CPU utilization, horizontal pod autoscaling will scale to according to the metric that requires the highest number of replicas.
+Kubernetes 1.6 adds supports for making use of custom metrics in the Horizontal Pod Autoscaler.
+You can custom metrics for the Horizontal Pod Autoscaler to use in the `autoscaling/v2alpha1` API.
+Kubernetes then queries the new custom metrics API to fetch the values of the appropriate custom metrics.
 
-If you do not specify a target for CPU utilization, Kubernetes defaults to an 80% utilization threshold for horizontal pod autoscaling.
+# Prerequisites
 
-If you want to ensure that horizontal pod autoscaling calculates the number of required replicas based only on custom metrics, you should set the CPU utilization target to a very large value (such as 100000%). As this level of CPU utilization isn't possible, horizontal pod autoscaling will calculate based only on the custom metrics (and min/max limits).
+In order to use custom metrics in the Horizontal Pod Autoscaler, you must have deployed the cluster with the
+`--horizontal-pod-autoscaler-use-rest-clients` flag on the controller manager set to true.  You must then configure
+your controller manager to speak to the API server through the API server aggregator.  The resource metrics API and
+custom metrics API must also be registered with the API server aggregator, and must be served by API servers running
+on the cluster.
 
+Currently, Heapster provides an implementation of the resource metrics API when run with the `--api-server` flag set
+to true. A separate component must provide the custom metrics API (more information on the custom metrics API is
+available at [the k8s.io/metrics repository](https://github.com.com/kubernetes/metrics)).
```

Comment:
It would be better to point to a particular document building pipeline about custom metrics rather than to the whole repository. If we don't have such document, we should as @piosz to write it.